sdv/multi_table/hma.py

```diff
@@ -260,12 +260,13 @@ def _extract_parameters(self, parent_row, table_name, foreign_key):
             num_rows = flat_parameters[num_rows_key]
             flat_parameters[num_rows_key] = min(
                 self._max_child_rows[num_rows_key],
-                math.ceil(num_rows)
+                max(0, round(num_rows))
```

Comment:
If we use math.ceil we can't get 0, because the synthesized data is almost always a small positive value.

I can move this out of this PR if we want to write test cases for this.

Reply:
I'm not sure if we need at least 1 here. I think there was a bug when we don't sample any child. If integration tests pass it should be good since it should be covered in there.

Reply:
If that value cannot be 0, then it's impossible for HMA to ever sample 0 rows for a primary key. For example, imagine you have two tables, parent and child, where the parent has 100 primary keys, but the every row of the child table only uses the first pk of the parent (so the foreign keys all have the same value). Currently, the sampled data will not be able to learn that, and instead every parent pk will be assigned to at least one child row.